cli/dts/lib.deno.unstable.d.ts

```diff
@@ -1108,6 +1108,92 @@ declare namespace Deno {
    * Make the timer of the given id not blocking the event loop from finishing
    */
   export function unrefTimer(id: number): void;
+
+  export interface CommandOptions {
+    /** Arguments to pass to the process. */
+    args?: string[];
+    /**
+     * The working directory of the process.
+     * If not specified, the cwd of the parent process is used.
+     */
+    cwd?: string | URL;
+    /**
+     * Clear environmental variables from parent process.
+     * Doesn't guarantee that only `opt.env` variables are present,
+     * as the OS may set environmental variables for processes.
+     */
+    clearEnv?: boolean;
+    /** Environmental variables to pass to the subprocess. */
+    env?: Record<string, string>;
+    /**
+     * Sets the child process’s user ID. This translates to a setuid call
+     * in the child process. Failure in the setuid call will cause the spawn to fail.
+     */
+    uid?: number;
+    /** Similar to `uid`, but sets the group ID of the child process. */
+    gid?: number;
+  }
+
+  export interface ProcessIoOptions {
+    stdin?: "piped" | "inherit" | "null";
+    stdout?: "piped" | "inherit" | "null";
+    stderr?: "piped" | "inherit" | "null";
+  }
+
+  /** **UNSTABLE** New API, yet to be vetted.
+   *
+   * Spawns new subprocess.
+   */
+  export class Command {
+    /**
+     * @param command The path to the binary to run.
+     * @param options Options for the Command.
+     */
+    constructor(command: string | URL, options?: CommandOptions);
+
+    /**
+     * Starts the command as a child process.
+     * By default, the io options are all set to `inherit`.
+     */
+    spawn<T extends ProcessIoOptions = ProcessIoOptions>(options?: T): Child<T>;
+    /**
+     * Executes the command as a child process, waiting for it to finish and collecting its exit status.
+     * By default, `stdout` and `stderr` are set to `inherit`.
+     */
+    status(options?: {
+      stdout?: "inherit" | "null";
+      stderr?: "inherit" | "null";
+    }): Promise<ProcessStatus>;
+    /** Executes the command as a child process, waiting for it to finish and collecting all of its output. */
+    output(): Promise<CommandOutput<{ stdin: "piped"; stderr: "piped" }>>;
+  }
+
+  export class Child<T extends ProcessIoOptions = ProcessIoOptions> {
+    readonly stdin: T["stdin"] extends "piped" ? WritableStream<Uint8Array>
+      : null;
+    readonly stdout: T["stdout"] extends "piped" ? ReadableStream<Uint8Array>
+      : null;
+    readonly stderr: T["stderr"] extends "piped" ? ReadableStream<Uint8Array>
+      : null;
+
+    readonly pid: number;
+    /** Get the current status. */
+    readonly status: ProcessStatus | null;
+
+    /** Waits for the child to exit completely, returning the status that it exited with. */
+    wait(): Promise<ProcessStatus>;
```

Comment:
What about `tryWait()` method that checks if process did exit but is synchronous?

Reply:
thats the `status` getter